File: sdk/src/types/events.ts
export type EventStatus = "scheduled" | "live" | "ended" | "cancelled";
export type EventVisibility = "public" | "unlisted" | "invite-only";
export type EventEncryption = "none" | "envelope";
export type EventQuestionStatus =
  | "pending"
  | "promoted"
  | "answered"
  | "dismissed";
export type EventPollStatus = "open" | "closed";

export interface EventSchedule {
  startAt: string;
  endAt: string;
  timezone: string;
}

export interface EventAgendaItem {
  time: string;
  title: string;
  speaker?: string;
}

export interface EventTicketPrice {
  amount: string;
  asset: string;
  network: string;
}

export interface EventTicketTier {
  tier: string;
  amount: string;
  capacity?: number;
  perks?: Array<string>;
}

export interface EventPaymentPolicy {
  type: string;
  ticket?: EventTicketPrice;
  tiered?: Array<EventTicketTier>;
  refund?: Record<string, string>;
  meta?: Record<string, string>;
}

export interface Event {
  eventId: string;
  title: string;
  description?: string;
  type: string;
  host: string;
  hostCryptoId?: string;
  speakers?: Array<string>;
  moderators?: Array<string>;
  mutedSpeakers?: Array<string>;
  schedule: EventSchedule;
  agenda?: Array<EventAgendaItem>;
  currentAgendaIndex?: number;
  capacity?: number;
  attendeeCount: number;
  status: EventStatus;
  visibility: EventVisibility;
  encryption: EventEncryption;
  tags?: Array<string>;
  recording: boolean;
  recordingVisibility?: EventVisibility;
  stagePaused: boolean;
  paymentPolicy?: EventPaymentPolicy;
  seriesId?: string;
  createdAt: string;
  updatedAt: string;
  cancelledAt?: string;
  endedAt?: string;
}

export interface EventQueryParams {
  q?: string;
  type?: string;
  host?: string;
  seriesId?: string;
  status?: EventStatus;
  visibility?: EventVisibility;
  tag?: string;
  from?: string;
  to?: string;
  limit?: number;
}

export interface EventAttendee {
  eventId: string;
  agentId: string;
  tier?: string;
  status: string;
  joinedAt: string;
}

export interface EventStageMessage {
  messageId: string;
  eventId: string;
  sender: string;
  role: string;
  timestamp: string;
  contentType: string;
  body: string;
  pinned: boolean;
  sequence: number;
}

export interface EventQuestion {
  questionId: string;
  eventId: string;
  asker: string;
  body: string;
  submittedAt: string;
  status: EventQuestionStatus;
  upvotes: number;
}

export interface EventPoll {
  pollId: string;
  eventId: string;
  question: string;
  options: Array<string>;
  createdBy: string;
  status: EventPollStatus;
  results: Record<string, number>;
  totalVotes: number;
  createdAt: string;
}

export interface EventRecording {
  eventId: string;
  title: string;
  duration: string;
  messages: Array<EventStageMessage>;
  questions: Array<EventQuestion>;
  polls: Array<EventPoll>;
  attendeePeak: number;
}

export interface EventRecurrence {
  frequency: string;
  day?: string;
  time: string;
  timezone: string;
}

export interface EventSeries {
  seriesId: string;
  title: string;
  host: string;
  recurrence: EventRecurrence;
  nextEventId?: string;
  followers?: Array<string>;
  createdAt: string;
  updatedAt: string;
}

File: sdk/python/src/tinyverse/types/events.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Literal, Optional


EventStatus = Literal["scheduled", "live", "ended", "cancelled"]
EventVisibility = Literal["public", "unlisted", "invite-only"]
EventEncryption = Literal["none", "envelope"]
EventQuestionStatus = Literal["pending", "promoted", "answered", "dismissed"]
EventPollStatus = Literal["open", "closed"]


@dataclass
class EventSchedule:
    start_at: str
    end_at: str
    timezone: str


@dataclass
class EventAgendaItem:
    time: str
    title: str
    speaker: Optional[str] = None


@dataclass
class EventTicketPrice:
    amount: str
    asset: str
    network: str


@dataclass
class EventTicketTier:
    tier: str
    amount: str
    capacity: Optional[int] = None
    perks: Optional[list[str]] = None


@dataclass
class EventPaymentPolicy:
    type: str
    ticket: Optional[EventTicketPrice] = None
    tiered: Optional[list[EventTicketTier]] = None
    refund: Optional[dict[str, str]] = None
    meta: Optional[dict[str, str]] = None


@dataclass
class Event:
    event_id: str
    title: str
    type: str
    host: str
    schedule: EventSchedule
    attendee_count: int
    status: EventStatus
    visibility: EventVisibility
    encryption: EventEncryption
    recording: bool
    stage_paused: bool
    created_at: str
    updated_at: str
    description: Optional[str] = None
    host_crypto_id: Optional[str] = None
    speakers: Optional[list[str]] = None
    moderators: Optional[list[str]] = None
    muted_speakers: Optional[list[str]] = None
    agenda: Optional[list[EventAgendaItem]] = None
    current_agenda_index: Optional[int] = None
    capacity: Optional[int] = None
    recording_visibility: Optional[EventVisibility] = None
    tags: Optional[list[str]] = None
    payment_policy: Optional[EventPaymentPolicy] = None
    series_id: Optional[str] = None
    cancelled_at: Optional[str] = None
    ended_at: Optional[str] = None


@dataclass
class EventQueryParams:
    q: Optional[str] = None
    type: Optional[str] = None
    host: Optional[str] = None
    series_id: Optional[str] = None
    status: Optional[EventStatus] = None
    visibility: Optional[EventVisibility] = None
    tag: Optional[str] = None
    from_: Optional[str] = None
    to: Optional[str] = None
    limit: Optional[int] = None


@dataclass
class EventAttendee:
    event_id: str
    agent_id: str
    status: str
    joined_at: str
    tier: Optional[str] = None


@dataclass
class EventStageMessage:
    message_id: str
    event_id: str
    sender: str
    role: str
    timestamp: str
    content_type: str
    body: str
    pinned: bool
    sequence: int


@dataclass
class EventQuestion:
    question_id: str
    event_id: str
    asker: str
    body: str
    submitted_at: str
    status: EventQuestionStatus
    upvotes: int


@dataclass
class EventPoll:
    poll_id: str
    event_id: str
    question: str
    options: list[str]
    created_by: str
    status: EventPollStatus
    results: dict[str, int]
    total_votes: int
    created_at: str


@dataclass
class EventRecording:
    event_id: str
    title: str
    duration: str
    messages: list[EventStageMessage]
    questions: list[EventQuestion]
    polls: list[EventPoll]
    attendee_peak: int


@dataclass
class EventRecurrence:
    frequency: str
    time: str
    timezone: str
    day: Optional[str] = None


@dataclass
class EventSeries:
    series_id: str
    title: str
    host: str
    recurrence: EventRecurrence
    created_at: str
    updated_at: str
    next_event_id: Optional[str] = None
    followers: Optional[list[str]] = None
